### PR TITLE
e2e test pr_bot, codebuild image and sample test

### PR DESCRIPTION
**Pull request by jaymccon**
## Overview

Adds e2e testing requirements:

* Codebuild image - tests are to be run in codebuild, custom image provides python 3.6.9 and 3.7.4 via pyenv.
* pr_bot - lambda function that is scheduled to poll the repo for pr comments which contain the tag `/do-e2e-tests`. Comment must come from a user that is listed as a maintainer in the `PROJECT_MAINTAINERS` env var.
* initial e2e test that checks whether taskcat cli is able to output help messages without failing.

will be testing the functionality here by putting in comments.

### CODE REVIEW

**Comment by avattathil** [comment]
Nice work on this @jaymccon

File: e2e/codebuild/run-e2e.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+FAILED=0
+
+function failed() {
+    FAILED=1
+    echo "TEST FAILED: ${t} python version: ${ver}"
+}
+
+for ver in "$@" ; do
+    pyenv global ${ver}
+    for t in $(ls -1 *.sh) ; do
```
Comment (andrew-glenn) [inline e2e/codebuild/run-e2e.sh]: This looks ripe for unintended results. Are you *sure* you want to do this?

File: e2e/pr_bot/handler.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+
+def get_pending_builds(history):
+    comments = {}
+    repo = Github(get_token()).get_repo(GITHUB_REPO_NAME)
+    prs = repo.get_pulls(state="open")
+    for pr in prs:
+        if pr.base.ref != "master":
+            continue
+        for c in pr.get_issue_comments():
```
Comment (andrew-glenn) [inline e2e/pr_bot/handler.py]: Readability:
`c` => `comment`

File: e2e/codebuild/run-e2e.sh (hunk 1 above)
Comment (jaymccon) [reply e2e/codebuild/run-e2e.sh]: elaborate?

Comment (andrew-glenn) [reply e2e/codebuild/run-e2e.sh]: Nah, fam

**Comment by jaymccon** [comment]
closing this due to all the testing comments, will re-open

**Pull request closed without merging**